File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] h) {
        int l = 0, r = h.length - 1;
        int ans = 0;
        while(l < r){
            int ln = Math.min(h[l], h[r]);
            int bd = r - l;
            ans = Math.max(ans, ln * bd);
            if(h[l] < h[r]) l++;
            else r--;
        }
        return ans;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        ans = -1
        while left < right:
            w = right - left
            l = min(height[left], height[right])
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
            ans = max(ans, l * w)
        return ans